### PR TITLE
[KV Connector] Keep MooncakeStore full hits block-aligned

### PR DESCRIPTION
**Pull request by Dao007forever**
## Summary
- round full-prompt MooncakeStore hits down to a block boundary before scheduling async load
- remove the downstream `+1` compensation paths in scheduler and worker now that `kvpool_cached_tokens` stays block-aligned
- make MooncakeStore always use block-aligned partial-chunk discard behavior by removing its private `_discard_partial_chunks` config path
- add regression coverage for the full-hit case where the final sampling tail is recomputed locally instead of loaded as a partial block

## Root cause
When MooncakeStore hit the entire prompt, the scheduler previously reduced the external hit from `num_tokens` to `num_tokens - 1` so the core scheduler would recompute the final token and produce logits for sampling. That made `kvpool_cached_tokens` sub-block aligned.

The recv side is block/chunk based: its load mask is built from block-aligned lengths, while `process_tokens()` can still yield a partial tail. That mismatch can either skip the tail silently or leave `key_list` empty when local cache already covers the aligned prefix.

This fix keeps the connector load boundary block-aligned by rounding the full hit down to `(num_tokens - 1) // block_size * block_size`. The final block is recomputed locally, which also avoids the linear/chunked-local attention boundary case. MooncakeStore now always follows this block-aligned partial-chunk behavior instead of carrying a separate `discard_partial_chunks` scheduler option.

## Why this is not a duplicate
- No linked issue number was provided for this branch, so there was no issue-specific `gh issue view <issue_number>` check to run.
- Checked open PRs in `vllm-project/vllm` with searches including `MooncakeStore full hit`, `MooncakeStore kvpool_cached_tokens`, `MooncakeStore block boundary`, `MooncakeStore round sampling-tail decrement`, and `KVCacheStoreRecvingThread key_list empty`.
- The nearby open PR #42788 propagates MooncakeStore load failures; it does not change full-hit load length or block-aligned final-token recompute behavior.

## Tests
- `/Users/daole/git/vllm-dao/.venv/bin/python -m pytest tests/v1/kv_connector/unit/test_mooncake_store_scheduler.py -v`
  - Result: all 10 test bodies passed, then local Mac/MPS teardown failed in `torch.accelerator.empty_cache()`.
- Directly invoked all 10 tests in `tests/v1/kv_connector/unit/test_mooncake_store_scheduler.py` from the updated PR branch.
  - Result: passed.
- `/Users/daole/git/vllm-dao/.venv/bin/python -m ruff check vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/scheduler.py tests/v1/kv_connector/unit/test_mooncake_store_scheduler.py`
  - Result: passed.

## AI assistance
AI assistance was used to inspect the MooncakeStore scheduler/worker paths, update the fix, add tests, and prepare this PR description. The human submitter is responsible for reviewing every changed line and defending the change end-to-end.

### CODE REVIEW

File: vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/scheduler.py [hunk 1]
```diff
@@ -341,10 +346,6 @@ def build_connector_meta(
                 if not load_spec:
                     continue
                 num_tokens_to_compute = load_spec.kvpool_cached_tokens
-                if (num_tokens_to_compute % self._block_size != 0) and (
-                    num_tokens_to_compute == unfinished_req.num_tokens - 1
-                ):
-                    num_tokens_to_compute = num_tokens_to_compute + 1
```
Comment (ivanium) [inline vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/scheduler.py:347]: We should fix this logic by getting the block back rather than removing it directly?

File: vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/scheduler.py [hunk 2]
```diff
@@ -99,7 +99,12 @@ def get_num_new_matched_tokens(
         num_external_hit_tokens = self.client.lookup(token_len, request.block_hashes)
 
         if num_external_hit_tokens == request.num_tokens:
-            num_external_hit_tokens -= 1
+            # Leave a sub-block tail uncomputed for sampling, on a block
```
Comment (ivanium) [inline vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/scheduler.py:92]: Can add an assertion in `__init__` saying that we only support `discard_partial_chunk`

File: vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/worker.py
Comment (ivanium) [inline vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/worker.py:1114]: same here. Should we fix the logic?

**Review by ivanium — COMMENTED** [review]
Valid issue. Left a few comments

File: vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/scheduler.py (hunk 2 above)
Comment (Dao007forever) [reply vllm/distributed/kv_transfer/kv_connector/v1/mooncake/store/scheduler.py:92]: Updated to remove discard_partial_chunk, and only support True.

**Review by ivanium — APPROVED** [review]
LGTM. It's a pity that we have to remove `discard_partial_chunks` for now, but we'd keep that in mind to add back in the future.

**Review by njhill — APPROVED** [review]

**Review by ywang96 — APPROVED** [review]
